ai2thor/controller.py

```diff
@@ -588,12 +616,29 @@ def key_sort_func(scene_name):
 
         # updates the initialization parameters
         self.initialization_parameters.update(init_params)
+
+        # RoboTHOR checks
+        agent_mode = self.initialization_parameters.get("agentMode", "default")
+        if agent_mode.lower() == "bot":
+            self.initialization_parameters["agentMode"] = "locobot"
+            warnings.warn("On reset and upon initialization, agentMode='bot' has been renamed to agentMode='locobot'.")
+        if (
+            scene in self.robothor_scenes() and
+            self.initialization_parameters.get("agentMode", "default").lower() != "locobot"
+        ):
+            warnings.warn("You are using a RoboTHOR scene without using the standard LoCoBot.\n" +
+             "Did you mean to mean to set agentMode='locobot' upon initialization or within controller.reset(...)?")
+
         self.last_event = self.step(
             action="Initialize", **self.initialization_parameters
         )
 
-        self.scene = scene
+        if not self.last_event.metadata['lastActionSuccess']:
```

Comment:
you can pass raise_for_failure=True to step('Initialize') and skip the lastActionSuccess check.

Reply:
Fixed in https://github.com/allenai/ai2thor/pull/605/commits/764cd71b4e2e3540049d19806d50b5d087654737